common/src/main/kotlin/com/hartwig/actin/molecular/panel/PanelSpecifications.kt

```diff
@@ -22,14 +22,8 @@ class PanelSpecifications(
         testSpec: PanelTestSpecification,
         negativeResults: Set<SequencedNegativeResult>?
     ): PanelTargetSpecification {
-        val unknownGenes = (negativeResults?.map(SequencedNegativeResult::gene)?.toSet() ?: emptySet()) - knownGenes
-        if (unknownGenes.isNotEmpty()) {
-            throw IllegalStateException(
-                "${logPanelName(testSpec)} has negative results associated containing " +
-                        "gene(s) not present in SERVE known genes: ${unknownGenes.joinToString()}." +
-                        "Correct this in the feed UI before continuing."
-            )
-        }
+
+        checkForUnknownGenesInNegativeResults(negativeResults, testSpec)
```

Comment:
You could consider wrapping this with `?.let` so you don't have to deal with nulls in the helper:

```suggestion
        negativeResults?.let { checkForUnknownGenesInNegativeResults(it, testSpec) }
```